usb/usb.py

```diff
@@ -0,0 +1,245 @@
+
+# Edit this function to ignore devices before the udev attributes are
+# computed in order to save time and memory.
+def fastIgnore(path):
+    # E.g. how to to ignore devices whose path begins with /dev/sda
+    #if path.startswith("/dev/sda"):
+    #    return True
+
+    # E.g. how to ignore a fixed set of paths
+    #if path in [ "/dev/path1", "/dev/path2", "/dev/path3" ]:
+    #    return True
+    return False
+
+###############################################################################
+# END CONFIG
+# DO NOT EDIT ANYTHING AFTER THIS POINT UNLESS YOU ARE A DEVELOPER
+###############################################################################
+
+from subprocess import check_output
+
+def pangoEscape(text):
+    return text.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
+
+def getLeafDevicePaths():
+    lines = check_output(['lsblk', '-spndo', 'NAME'], universal_newlines=True)
+    lines = lines.split("\n")
+    lines = filter(None, lines)
+    return lines
+
+def getKernelName(path):
+    return check_output(['lsblk', '-ndso', 'KNAME', path],
+            universal_newlines=True).rstrip("\n")
+
+def getDeviceType(path):
+    return check_output(['lsblk', '-no', 'TYPE', path],
+            universal_newlines=True).strip() 
+
+def getUnlockedCryptOutput(path):
+    return path
+
+def getFSType(path):
+    global attributeMaps
+    return attributeMaps[path].get("ID_FS_TYPE")
+
+def isLUKSPartition(path):
+    return getFSType(path) == "crypto_LUKS"
+
+def isSwapPartition(path):
+    return getFSType(path) == "swap"
+
+def getFSLabel(path):
+    global attributeMaps
+    label = attributeMaps[path].get("ID_FS_LABEL_ENC", "")
+    if label:
+        label = label.encode().decode("unicode-escape")
+        if type(TRUNCATE_FS_LABELS) == int:
+            if TRUNCATE_FS_LABELS >= 0:
+                label = label[:TRUNCATE_FS_LABELS]
+            elif TRUNCATE_FS_LABELS < 0:
+                label = label[TRUNCATE_FS_LABELS:]
+    return label
+
+def getFSOptions(path):
+    lines = check_output(['findmnt', '-no', 'FS-OPTIONS', path],
+            universal_newlines=True).strip()
+    lines = lines.split(",")
+    return lines
+
+def isReadOnly(path):
+    return "ro" in getFSOptions(path)
+
+def isExtendedPartitionMarker(path):
+    global attributeMaps
+    MARKERS = ["0xf", "0x5"]
+    return attributeMaps[path].get("ID_PART_ENTRY_TYPE") in MARKERS
+
+def getMountPoint(path):
+    return check_output(['lsblk', '-ndo', 'MOUNTPOINT', path],
+            universal_newlines=True).rstrip("\n")
+
+def getSpaceAvailable(path):
+    lines = check_output(['df', '-h', '--output=avail', path],
+            universal_newlines=True)
+    lines = lines.split("\n")
+    if len(lines) != 3:
+        return ""
+    else:
+        return lines[1].strip()
+
+def getLockedCryptOutput(path):
+    form = "<span color='{}'>[{} {}]</span>"
+    kname = getKernelName(path)
+    output = form.format(LOCKED_COLOR, FA_LOCK, kname)
```

Comment:
AFAIK `df -h --output=avail` will never output any character that needs to be escaped ( '&', '<', '>'). Should I guard against this anyway?